### ISSUE TITLE
File names in command line

### ISSUE THREAD

**Issue by ncarrut** [smith-chem-wisc/MetaMorpheus#1277]
Hi,
I'm using MetaMorpheus in command line mode and then parsing the results.  Can I set file names from the command line?  For example make the result folder "myResult" instead of the date stamp.  Also set the base result file name to get "nyName_Peptides.psmtsv" instead of the current result which is either aggregate_Peptides.psmtsv or the raw-file name _Peptides.psmtsv depending on how many files were searched.  
Nick

**Comment by rmillikin** [comment]
Hi Nick,

You can specify the output folder with -o. For example:

`.\CMD.exe -t "C:\Data\SlicedYeastRawFiles\2018-07-02-14-07-16\Task2-SearchTask\SearchTaskconfig.toml" -o "C:\Data\SlicedYeastRawFiles\MyOutputFolder" -s "C:\Data\SlicedYeastRawFiles\SmallCalibratibleYeast.mzml" -d "C:\Data\SlicedYeastRawFiles\smalldb.fasta"`

Currently there's no way to rename the tsv output files; we organize the output such that each file has a list of PSMs/proteins, and then the aggregated one with all the PSMs/proteins from all the files. I'm not quite sure how to organize the output with a custom name unless we append the spectra file name on top of the user-defined name. How do you want this to work? Do you just want the aggregate file with a custom name and the individual file results remain the same as current?

As a side note, you can give each task a name, if that helps you organize your results. The task's name is defined in its .toml.

Rob

**Comment by ncarrut** [comment]
Thanks for the response Rob.  I'm trying to make an automated search for QC
runs.  Sometimes it runs with just one .raw file and sometimes more.  One
file gives just the individual results named according to the raw file (eg
"myRawfile_PSMs.psmtsv").  Multiple files gives the aggregate file (eg.
"aggregate_PSMs.psmtsv").  I'm making my R script look for different file
names depending on whether I searched one .raw file or multiple and it
would make my life easier if there was just one file name to expect.  I see
your point about the awkwardness of a user-defined name.  Maybe you could
create the aggregate file regardless of the number of files searched.  Or
maybe I could just stick with searching for the raw-file based name when
appropriate.
Nick

**Comment by rmillikin** [comment]
I think that solution makes sense; always creating the aggregate output file (even with 1 spectra file searched) and giving the option of a user-defined name. And if >1 files are searched, the individual file results are additionally written with their respective spectra file names.

This hopefully shouldn't be too complex to implement but may take some time to implement. Not sure about an ETA - maybe 2-3 weeks?

**Comment by ncarrut** [comment]
That would be great!  As long as there is an aggregate file I don’t need the user-defined file names so I’m neutral about including that part.

**Comment by rmillikin** [comment]
This is pretty much done and will be included in the next release. We've just made the aggregate file always be output. We've also revised the names of the files and put the individual files in a subdirectory to clean things up a bit. Attached is an example.

So it'll break your current program (sorry) but the end result is much cleaner/simpler.

![capture](https://user-images.githubusercontent.com/24705713/43420772-6ef93be2-940a-11e8-9d04-904c802a9427.PNG)
